iroh/src/magicsock/metrics.rs

```diff
@@ -77,4 +77,88 @@ pub struct Metrics {
+    pub path_rtt_variance_ms: Histogram,
+    /// Histogram of path quality scores (0.0-1.0).
+    pub path_quality_score: Histogram,
+}
+
+impl Default for Metrics {
+    fn default() -> Self {
+        Self {
+            update_direct_addrs: Counter::default(),
+            send_ipv4: Counter::default(),
+            send_ipv6: Counter::default(),
+            send_relay: Counter::default(),
+            send_relay_error: Counter::default(),
+            send_data: Counter::default(),
+            send_data_network_down: Counter::default(),
+            recv_data_relay: Counter::default(),
+            recv_data_ipv4: Counter::default(),
+            recv_data_ipv6: Counter::default(),
+            recv_datagrams: Counter::default(),
+            recv_gro_datagrams: Counter::default(),
+            send_disco_udp: Counter::default(),
+            send_disco_relay: Counter::default(),
+            sent_disco_udp: Counter::default(),
+            sent_disco_relay: Counter::default(),
+            sent_disco_ping: Counter::default(),
+            sent_disco_pong: Counter::default(),
+            sent_disco_call_me_maybe: Counter::default(),
+            recv_disco_bad_key: Counter::default(),
+            recv_disco_bad_parse: Counter::default(),
+            recv_disco_udp: Counter::default(),
+            recv_disco_relay: Counter::default(),
+            recv_disco_ping: Counter::default(),
+            recv_disco_pong: Counter::default(),
+            recv_disco_call_me_maybe: Counter::default(),
+            recv_disco_call_me_maybe_bad_disco: Counter::default(),
+            relay_home_change: Counter::default(),
+            num_direct_conns_added: Counter::default(),
+            num_direct_conns_removed: Counter::default(),
+            num_relay_conns_added: Counter::default(),
+            num_relay_conns_removed: Counter::default(),
+            actor_tick_main: Counter::default(),
+            actor_tick_msg: Counter::default(),
+            actor_tick_re_stun: Counter::default(),
+            actor_tick_portmap_changed: Counter::default(),
+            actor_tick_direct_addr_heartbeat: Counter::default(),
+            actor_link_change: Counter::default(),
+            actor_tick_other: Counter::default(),
+            nodes_contacted: Counter::default(),
+            nodes_contacted_directly: Counter::default(),
+            connection_handshake_success: Counter::default(),
+            connection_became_direct: Counter::default(),
+            path_marked_outdated: Counter::default(),
+            path_ping_failures: Counter::default(),
+            path_failure_resets: Counter::default(),
+            path_packet_loss_rate: packet_loss_buckets(),
+            path_rtt_variance_ms: rtt_variance_buckets(),
+            path_quality_score: quality_score_buckets(),
+        }
+    }
+}
+
+fn packet_loss_buckets() -> Histogram {
+    Histogram::new(vec![0.0, 0.01, 0.05, 0.1, 0.2, 0.5, 1.0])
```

Comment:
I guess the buckets are:
- [0, 0.01)
- [0.01, 0.05)
- [0.1, 0.2)
- [0.2, 0.5)
- [0.5, 1.0)
- [1.0, infinity]

?